### ISSUE TITLE
Question about training time and GPT-based judge cost in your experiments

### ISSUE THREAD

**Issue by LimHyeongSoo** [qualidea1217/HiPRAG#5]
Hello,

I read your paper with great interest, and I found the proposed method and experimental results very impressive. I am currently trying to reproduce and implement your experiments to better understand the approach in practice.

From the paper, I understand that the main experiments were conducted using 4× NVIDIA A100 GPUs, and that a GPT-based judge model was used for evaluation. In order to better plan my reproduction setup, I would like to ask if you could share some additional practical details:

Approximate training time (e.g., wall-clock time) for the main experiments
Rough cost estimation or usage statistics for the GPT-based judge model (e.g., number of calls, tokens, or overall API cost), if available
I fully understand that exact numbers may vary or may not be precisely tracked, so even high-level or approximate information would be very helpful.

Thank you again for sharing this work, and I appreciate any insights you are able to provide.

**Comment by qualidea1217** [comment]
Hi, thank you for your interest in our work! You can expect around 48-72 hours for a full training. The specific time depends on your settings (for example, more gpus offer faster training). In terms of cost estimation, I am sorry that we have lost that part of the data so I can't give you an answer. I suggest you try a few steps first and calculate based on your own situation. You don't have to use gpt-based model though.

**Comment by LimHyeongSoo** [comment]
Thank you for your previous response.

I have a follow-up question regarding the training configuration reported in the paper.
In the paper, it is clearly stated that training was conducted using 4× A100 GPUs and terminated at 400 steps. However, in the training script provided in the repository — specifically:

/data/hslim/PycharmProject/HyperscaleES/HiPRAG/scripts/nq_hotpotqa/v0.3/train_ppo_format.sh

the number of training steps appears to be set to 1005.

Could you please clarify which configuration is correct?
Was the model in the paper trained for 400 steps as described, or for 1005 steps as indicated in the script?
If the script reflects a different experimental setting, it would be helpful to understand the distinction.

Thank you for your clarification.

**Comment by qualidea1217** [comment]
Hi, the model was trained with 400 steps. The 1005 steps is a setting inherited from one of the original Search-R1 training scripts that we don't use and sometimes we manually end the training at step 400. In reality we found that most of the training converges at or before 400 steps.

**Comment by LimHyeongSoo** [comment]
Then, in your README.md, you clearly wrote the following under Quick Start:

`Train a reasoning + search LLM with HiPRAG's hierarchical process rewards on NQ + HotpotQA dataset with e5 as the retriever and wikipedia as the corpus.`

And specifically:

5) Run RL training (PPO) with HiPRAG's reward system.

```
conda activate searchr1  
bash scripts/nq_hotpotqa/v0.3/train_ppo_format.sh
```

This suggests that running the above script would execute the HiPRAG system with your hierarchical reward mechanism.

However, are you now saying that this script file is actually a Search-R1 script rather than a HiPRAG-specific implementation?

If so, could you clarify whether this GitHub repository truly contains the implementation corresponding to your HiPRAG paper?

Is it sufficient to simply modify the script file, or are there additional components that are not publicly released?

Additionally, are the exact training scripts used in the paper not publicly available? The contents described in the README.md appear to differ from what is currently provided in the repository.

I would appreciate clarification on this point.

**Comment by LimHyeongSoo** [comment]
hank you for the clarification.

Based on your explanation, am I correct in understanding that we simply need to modify the training step parameter to 400 in the provided script?

In other words, when following the README instructions:


> Quick Start
> Train a reasoning + search LLM with HiPRAG's hierarchical process rewards on NQ + HotpotQA dataset with e5 as the retriever and wikipedia as the corpus.


```
conda activate searchr1  
bash scripts/nq_hotpotqa/v0.3/train_ppo_format.sh

```

should we interpret that this script indeed runs the HiPRAG system (with the hierarchical reward mechanism), and that the only adjustment needed to reproduce the paper’s setting is to change the total training steps to 400?

Please confirm whether this understanding is correct.

**Comment by qualidea1217** [comment]
The script runs the HiPRAG system, not the original Search-R1 method, as the related code in verl has changed to HiPRAG's method. However, depending on the specific hardware and software environment, you may need to do extra settings or tune other parameters. This is a problem largely due to the outdated original Search-R1 environment which has a lot of bugs and missing tons of useful features such as continue training from a checkpoint. For example, there was a time that we spent almost three weeks to get it work on one of our servers due to some weird problems related to Ray. Honestly, if I were asked to reproduce this paper, I would implement it on my own from a fresh start with a newer RL training framework instead of this one.
